core/trino-main/src/main/java/io/trino/server/security/oauth2/RefreshTokensConfig.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server.security.oauth2;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.jsonwebtoken.SignatureAlgorithm;
+
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+
+import java.time.Duration;
+
+public class RefreshTokensConfig
+{
+    private SignatureAlgorithm signingAlgorithm = SignatureAlgorithm.ES512;
+
+    private long refreshTokenExpirationExtensionSeconds = Duration.ofMinutes(10).getSeconds();
+
+    public SignatureAlgorithm getSigningAlgorithm()
+    {
+        return signingAlgorithm;
+    }
+
+    @Config("http-server.authentication.oauth2.refresh-tokens.signing.algorithm")
+    @ConfigDescription("Signature algorithm that will be used to secure a token. If no keys will be provided, it will be used as a base to generate adequate keys")
+    public RefreshTokensConfig setSigningAlgorithm(String algorithm)
```

Comment:
Yes, but one of the value of the SignatureAlgorithm is NONE. I think that I would like to prevent that from beeing used. (I've forget to add that here) WIll airlift print out all possible values if we use enum here? If not then I can use enum and make sure that it's never NONE.